utilities/Microsoft.Quantum.Katas/CheckKataMagic.cs

```diff
@@ -44,185 +44,76 @@ public CheckKataMagic(IOperationResolver resolver, ICompilerService compiler, IL
                     "```\n"
                 }
             };
-            this.Kind = SymbolKind.Magic;
-            this.Execute = this.Run;
-
-            this.Resolver = resolver;
             this.Compiler = compiler;
-            this.Logger = logger;
         }
 
-        /// <summary>
-        /// The Resolver lets us find compiled Q# operations from the workspace
-        /// </summary>
-        protected IOperationResolver Resolver { get; }
-
         /// <summary>
         /// The list of user-defined Q# code snippets from the notebook.
         /// </summary>
         protected ICompilerService Compiler { get; }
 
-        protected ILogger<KataMagic> Logger { get; }
-
         /// <summary>
-        /// What this Magic does when triggered. It will:
-        /// - find the Test to execute based on the provided name,
-        /// - semi-compile the code after to identify the name of the operation with the user's answer.
-        /// - call simulate to execute the test.
+        /// Semi-compiles the given code. Checks there is only one operation defined in the code,
+        /// and returns its corresponding OperationInfo
+        /// The compiler does this on a best effort basis, and in particular without relying on any context and/or type information,
+        /// so it will return the operation even if the compilation fails.
         /// </summary>
-        public virtual async Task<ExecutionResult> Run(string input, IChannel channel)
-        {
-            channel = channel.WithNewLines();
-
-            // Expect exactly two arguments, the name of the Kata and the user's answer (code).
-            var args = input?.Split(new char[] { ' ', '\n', '\t' }, 2);
-            if (args == null || args.Length != 2)
-            {
-                channel.Stderr("Invalid parameters. Usage: `%kata Test \"Q# operation\"`");
-                return ExecuteStatus.Error.ToExecutionResult();
-            }
-
-            var name = args[0];
-            var code = args[1];
-
-            var test = FindTest(name);
-            if (test == null)
-            {
-                channel.Stderr($"Invalid test name: {name}");
-                return ExecuteStatus.Error.ToExecutionResult();
-            }
-
-            var userAnswer = Compile(code, channel);
-            if (userAnswer == null) { return ExecuteStatus.Error.ToExecutionResult(); }
-
-            return Simulate(test, userAnswer, channel)
-                ? "Success!".ToExecutionResult()
-                : ExecuteStatus.Error.ToExecutionResult();
-        }
+        protected override QsNamespaceElement[] GetDeclaredCallables(string code, IChannel channel) =>
+               Compiler
+               .IdentifyElements(code)
+               .ToArray();
```

Comment:
```suggestion
               .IdentifyElements(code);
```

Reply:
Hi @anjbur. I unfortunately can't do this change. 

`ToArray()` method is being used to get the uniform return type from `GetDeclaredCallables(code, channel)`. <br> `GetDeclaredCallables(code, channel)` calls one of these methods according to the magics to compile or semi-compile  the code.
- **Semi-compilation in "CheckKataMagic" :** `Compiler.IdentifyElements(code)` returns `IEnumerable<QsNamespaceElement>`
- **Compilation in "KataMagic" :** `Snippets.Compile(code)` returns `QsNamespaceElement[]`

Reply:
Why not change the return type to `IEnumerable<QsNamespaceElement>` then? The result doesn't need to be an array to be used in `Compile`

Reply:
Hi @anjbur. Thanks for the clarification. I have now changed the return type to `IEnumerable<QsNamespaceElement>`.